src/js/vic-v2/components/PhotoField.jsx

```diff
@@ -231,31 +267,73 @@ export default class PhotoField extends React.Component {
     this.setState({ dragAndDropSupported: dragAndDropSupported && !iOS });
   }
 
-  detectWidth = () => {
-    const windowWidth = window.innerWidth;
-    this.setState({ windowWidth });
+  updateBoundaryWarningAndButtonStates = () => {
+    const canvasCropBoundariesMet = this.getCanvasCropBoundariesMet(this.refs.cropper.getCanvasData(), this.refs.cropper.getCropBoxData());
+
+    this.updateMoveButtonsEnabledStates(canvasCropBoundariesMet);
+    this.maybeShowCropBoundaryWarning(canvasCropBoundariesMet);
+  }
+
+  updateMoveButtonsEnabledStates = ({ topBoundaryMet, bottomBoundaryMet, rightBoundaryMet, leftBoundaryMet }) => {
+    const buttonStates = {
+      moveUpDisabled: bottomBoundaryMet,
+      moveDownDisabled: topBoundaryMet,
+      moveLeftDisabled: rightBoundaryMet,
+      moveRightDisabled: leftBoundaryMet
+    };
+
+    this.setState(buttonStates);
+  }
+
+  makeBoundaryEdgeWarning = (direction) => `You have reached the edge of your photo and cannot move it any farther ${direction}. Use the other arrows or the make larger and make smaller buttons to continue to edit.`
+
+  maybeShowCropBoundaryWarning = ({ zoomWarn = false, topBoundaryMet, bottomBoundaryMet, rightBoundaryMet, leftBoundaryMet }) => {
```

Comment:
Complexity of this kind of grew with the addition of the warnings. Wanted all of the logic controlling the determination of the the warning to display to be in one place.